### PR TITLE
Display chapter index instead of percentage if applicable

### PR DESCRIPTION
**Pull request by pichfl**
Fixing a few iOS 12 related things along the way. I should have split this into two PRs but I butchered the split when I started making changes to the storyboard.

Re #154

### CODE REVIEW

File: BookPlayer/Library/BaseListViewController.swift [hunk 1]
```diff
@@ -213,21 +214,20 @@ extension BaseListViewController: UITableViewDataSource {
...
         cell.isPlaylist = item is Playlist
-        cell.playbackState = .Stopped
-        cell.titleColor = UIColor.black
+        cell.playbackState = .stopped
 
         if let book = item as? Book {
             cell.subtitle = book.author
 
-            cell.progress = item.percentCompleted / 100.0
+            cell.progress = item.currentTime / item.duration
```
Comment (GianniCarlo) [inline BookPlayer/Library/BaseListViewController.swift]: you could use the introduced `progress` calculated var in the `Book` class

**Comment by GianniCarlo** [comment]
We need to do something about that truncated text.
img_1356
Maybe we could change the constraints, I see that the horizontal space is being split equally among the three elements (current time, progress, max time).

We could make it so the `current time` and `max time` only take up horizontal space to accommodate this string `000:00`, and that would free up a lot of space for the `progress` label

File: BookPlayer/Player/PlayerManager.swift [hunk 2]
```diff
@@ -95,24 +97,28 @@ class PlayerManager: NSObject {
...
-        guard let audioplayer = self.audioPlayer else {
+        guard let audioplayer = self.audioPlayer, let book = self.currentBook else {
             return
         }
 
-        self.currentBook.currentTime = audioplayer.currentTime
+        book.currentTime = audioplayer.currentTime
 
-        let isPercentageDifferent = self.currentBook.percentage != self.currentBook.percentCompleted
+        let isPercentageDifferent = book.percentage != book.percentCompleted || (book.percentCompleted == 0 && book.progress > 0)
```
[the remark this answers is not in the document]
Comment (pichfl) [reply BookPlayer/Player/PlayerManager.swift]: Hmmm, will have to check again

**Comment by pichfl** [comment]
Hmm, we need to support labels up to 00:00:00 on both sides even on legacy devices as the SE as single files can potentially last for up to 13 hours. I would try to abbreviate or even drop the word "Chapter", both Chpt. or Ch. are commonly used. I'm going to try to find a middle ground that fits all of these.

**Comment by GianniCarlo** [comment]
right `00:00:00` is what I meant haha, current book I'm listening to has the chapters split underneath the hour mark so I got that mixed up

File: BookPlayer/Library/BaseListViewController.swift (hunk 1 above)
Comment (pichfl) [reply BookPlayer/Library/BaseListViewController.swift]: Didn't have the property when I started, but I should definitely do that :D

File: BookPlayer/Player/PlayerManager.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (pichfl) [reply BookPlayer/Player/PlayerManager.swift]: Yes, this is a safeguard to trigger the notification for changes smaller than 1% when starting a book.

**Pull request merged**